src/js/view/controls/components/timeslider.js

```diff
@@ -286,6 +294,27 @@ class TimeSlider extends Slider {
         }
     }
 
+    setAriaText(position, duration) {
+        if (this.streamType === 'DVR') {
+            this.ariaText = timeFormat(position);
+        } else {
+            this.ariaText = `${timeFormat(position)} of ${timeFormat(duration)}`;
+        }
+    }
+
+    updateAriaText(evt) {
+        if (evt && evt.type === 'keydown' &&
+            evt.key !== 'ArrowLeft' &&
+            evt.key !== 'ArrowRight') {
+            return;
+        }
+
+        // wait for seek to complete before updating aria text
+        setTimeout(() => {
```

Comment:
If you add a timeout you have to cancel it when appropriate.

This doesn't make sense. A timeout doesn't not mean a seek completed. If you need to update a value based on change, use the model.